pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -827,6 +835,69 @@ allChainsLabel:
 	return mutable.Listener
 }
 
+func buildSidecarListenerTLSContext(tlsOpts *networking.Server_TLSOptions) *auth.DownstreamTlsContext {
+	// ingress inbound tls only support "SIMPLE" and "MUTUAL" mode
+	if tlsOpts == nil || (tlsOpts.Mode != networking.Server_TLSOptions_SIMPLE && tlsOpts.Mode != networking.Server_TLSOptions_MUTUAL) {
```

Comment:
Should we log a warning/add a metric here that it is not a valid Mode? And as a side note, should the sidecar validation function be enhanced with this new setting and validate correctly there?

Reply:
yes, i have validated there. This is just in case.